tests/test_imputation/test_tmi_imputation.py

```diff
@@ -0,0 +1,130 @@
+import pandas as pd
+from pandas.testing import assert_frame_equal
+import pytest
+
+from src.imputation.tmi_imputation import (
+    create_mean_dict,
+    apply_tmi,
+)
+
+
+def test_create_mean_dict():
+
+    # Create a sample DataFrame for testing
+    data = {
+        'statusencoded': ["210", "210", "210", "210", "210", "210", "210",
+                          "210", "210", "210", "211", "210", "211", "210",
+                          "211", "not_210_or_211"],
+        'imp_class': ['A', 'A', 'A', 'A', 'A', 'A', 'A', 'A', 'A', 'A', 'A',
+                      'B', 'B', 'C', 'C', 'C'],
+        'Value_trim': ['dont trim', 'dont trim', 'dont trim', 'dont trim',
+                       'dont trim', 'dont trim', 'dont trim', 'dont trim',
+                       'dont trim', 'dont trim', 'dont trim', 'dont trim',
+                       'dont trim', 'dont trim', 'dont trim', 'dont trim'],
+        'Value1': [100, 100, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1], #TODO if both same which should be trimmed?
```

Comment:
good point on the to do: if values are the same, we order by employees then reference, so in practice there shouldn't be ambiguity.

We could add "instance" to the sort if you think it would be needed.